src/main/java/com/todoist/pojo/ViewOption.kt

```diff
@@ -1,25 +1,47 @@
 package com.todoist.pojo
 
+import java.util.Locale
+
 open class ViewOption(
     id: Long,
-    open var viewType: Type?,
+    open var viewType: Type,
     open var objectId: Long?,
     open var sortedBy: Sort?,
     open var sortOrder: SortOrder?,
     open var groupedBy: Group?,
     open var filteredBy: String?,
     isDeleted: Boolean
 ) : TodoistObject(id, isDeleted) {
-    enum class Type(private val key: String) {
-        TODAY("TODAY"),
-        PROJECT("PROJECT"),
-        LABEL("LABEL"),
-        FILTER("FILTER");
+    sealed class Type(protected open val key: String) {
+        object Today : Type(KEY_TODAY)
 
-        override fun toString() = key
+        object Project : Type(KEY_PROJECT)
+
+        object Label : Type(KEY_LABEL)
+
+        object Filter : Type(KEY_FILTER)
+
+        data class Unknown(override val key: String) : Type(key)
+
+        object Invalid : Type(KEY_INVALID)
+
+        override fun toString() = key.toUpperCase(Locale.getDefault())
 
         companion object {
-            fun get(typeKey: String?): Type? = values().find { it.key == typeKey }
+            private const val KEY_TODAY = "TODAY"
+            private const val KEY_PROJECT = "PROJECT"
+            private const val KEY_LABEL = "LABEL"
+            private const val KEY_FILTER = "FILTER"
+            private const val KEY_INVALID = "INVALID"
+
+            fun get(typeKey: String?) = when {
+                typeKey.isNullOrEmpty() -> Invalid
+                typeKey.equals(KEY_TODAY, true) -> Today
```

Comment:
Two thoughts:
1. Do we need the case-insensitive comparison? In other API responses, we do expect a particular letter case.
2. We could inline the string keys when initializing the objects (like we do in the other enums) and use the object's `key` here? Actually don't feel strongly either way about using the constants or not, but using `{object}.key` looks simpler to me.

So e.g. `typeKey == Today.key -> Today`.

Reply:
> Do we need the case-insensitive comparison?

I was mostly playing it safe, trying to minimize the error prone-ness (? 🤔). That said, and looking at both points of your suggestion, makes sense to discard this and keeping it simple with the key comparison 👍